packages/core/contracts/UnirepSocial.sol

```diff
@@ -368,29 +410,65 @@ contract UnirepSocial is zkSNARKHelper {
         uint256[8] memory proof
     ) external payable {
         (,,,,uint maxReputationBudget,,,uint attestingFee,,) = unirep.config();
+        uint256 epochKey = publicSignals[0];
+        require(publicSignals[maxReputationBudget + 4] == commentReputation, "Unirep Social: submit different nullifiers amount from the required amount for comment");
         // check if proof is submitted before
         bytes32 proofNullifier = keccak256(
             abi.encodePacked(publicSignals, proof)
         );
         require(!usedProofNullifier[proofNullifier], "Unirep Social: the proof is submitted before");
         usedProofNullifier[proofNullifier] = true;
         require(publicSignals[maxReputationBudget + 3] == attesterId, "Unirep Social: submit a proof with different attester ID from Unirep Social");
+        require(postId < contentId, "Unirep Social: the post id is not valid");
 
         uint256 epoch = publicSignals[maxReputationBudget + 2];
-        uint256 epochKey = publicSignals[0];
         uint256 proofSpendAmount = publicSignals[maxReputationBudget + 4];
         require(proofSpendAmount == commentReputation, "Unirep Social: submit different nullifiers amount from the required amount for comment");
 
         // Spend reputation
         unirep.spendReputation{value: attestingFee}(publicSignals, proof);
 
+        // saved post id and hashed content
+        uint256 commentId = contentId;
+        hashedContentMapping[commentId][contentHash] = epochKey;
+
         emit CommentSubmitted(
             epoch,
             postId,
-            epochKey, // epoch key
+            epochKey,
+            commentId,
             contentHash,
             publicSignals[maxReputationBudget + 5] // min rep
         );
+
+        // update comment Id
+        contentId ++;
+    }
+
+    /*
+     * Update a published post/comment content
+     * @param id The post ID or the comment ID
+     * @param oldContentHash The old hashed content of the post/comment
+     * @param newContentHash The new hashed content of the post/comment
+     * @param publicSignals The public signals of the epoch key proof of the author of the post/comment
+     * @param proof The epoch key proof of the author of the post/comment
+     */
+    function edit(
+        uint256 id,
+        bytes32 oldContentHash,
+        bytes32 newContentHash,
+        uint256[] memory publicSignals,
```

Comment:
We really need a signing system for epoch keys, like semaphore has. The user should prove their epoch key and the `id`/`oldContentHash`/`newContentHash`. Then the backend can't modify the content.

We can do this in a different pr though.